### PR TITLE
[DOCS] Add documentation for near real-time search

### LINKED ISSUE #40644: Add up-to-date docs for near-realtime search
The very first [basic concept](https://www.elastic.co/guide/en/elasticsearch/reference/current/getting-started-concepts.html) is that Elasticsearch is a near-realtime NRT search platform, but this is not defined in more depth in the reference manual. We receive questions (with low but nonzero frequency) of the form "I wrote a document and got an ack and then tried to search for it and it wasn’t there" and it'd be good if there was somewhere in the docs that answers this question so we could just share a link.

Today our options are:

- the [relevant page of the definitive guide](https://www.elastic.co/guide/en/elasticsearch/guide/current/near-real-time.html) - this page contains good information, but the definitive guide as a whole is not up to date so it's risky to link new users there.

- the [docs for the `?refresh` option](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-refresh.html) covers, or at least mentions, most of the ideas, except for...

- the [`POST _refresh` API](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-refresh.html), but by default this is usually unnecessary because...

- [`index.refresh_interval`](https://www.elastic.co/guide/en/elasticsearch/reference/current/index-modules.html#dynamic-index-settings) defaults to `1s`.

I think it'd be great if we could rework this into a single place in the reference docs that describes what NRT means as a concept, and how refreshes happen in Elasticsearch. Although it isn't really anything to do with replication, the [replication overview](https://www.elastic.co/guide/en/elasticsearch/reference/6.7/docs-replication.html) is really more about the read and write models and I think that documenting the visibility semantics could fit here.

Additionally, it's not very clear that update-by-query and delete-by-query rely on searches, and therefore depend on which documents have been refreshed.

/cc @elastic/docs

### PR DESCRIPTION
**Pull request by lockewritesdocs**
This PR adds documentation explaining [near real-time search](https://elasticsearch_57560.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/near-real-time.html) to the ES Reference. 

Closes [#40644](https://github.com/elastic/elasticsearch/issues/40644)

Backports:
* [7.8] #58131
* [7.7] #58133
* [7.x] #58138

### CODE REVIEW

File: docs/reference/search/near-real-time.asciidoc [hunk 1]
```diff
@@ -0,0 +1,21 @@
+[[near-real-time]]
+== Near real-time search
```
Comment (DaveCTurner) [inline docs/reference/search/near-real-time.asciidoc:2]: Can we lift some of the content/diagrams from the definitive guide to this page? I think these are difficult concepts for beginners to handle, so we should try and start from the basics and make this page as self-contained as possible. The target audience includes first-time users that just started Elasticsearch, indexed some docs, tried to search for them, and didn't find them.

File: docs/reference/search/near-real-time.asciidoc [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+The overview of <<documents-indices,documents and indices>> indicates that when a document is stored in {es}, it is indexed and fully searchable in _near real-time_--within 1 second. What defines near real-time search?
```
Comment (DaveCTurner) [inline docs/reference/search/near-real-time.asciidoc:3]: Could you wrap to 80 columns?

File: docs/reference/search/near-real-time.asciidoc [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+Lucene, the Java libraries on which {es} is based, introduced the concept of per-segment search. A _segment_ is similar to an inverted index, but the word _index_ in Lucene means "a collection of segments plus a commit point". After a commit, a new segment is added to the commit point and the buffer is cleared.
+
+Lucene allows new segments to be written and opened, making the documents they contain visible to search ​without performing a full commit. This is a much lighter process than a commit to disk, and can be done frequently without degrading performance.
+
+In {es}, this lightweight process of writing and opening a new segment is called a _refresh_. A refresh makes all operations performed on an index since the last refresh available for search. By default, every shard is refreshed automatically once per second. This is why we say that {es} has _near_ real-time search: document changes are not visible to search immediately, but will become visible within 1 second.
```
Comment (DaveCTurner) [inline docs/reference/search/near-real-time.asciidoc:9]: Could we expand this to spell out all the different ways that a refresh might happen?

- wait for the refresh interval
- use the `?refresh` option
- perform an explicit `POST _refresh`

Also let's not call it "lightweight" as an absolute term. It's certainly lighter than a full commit, but it's not free and you can cause performance issues by doing it too much.

Comment (DaveCTurner) [inline docs/reference/search/near-real-time.asciidoc:9]: Also, the default isn't quite to refresh every second, it's more complicated than that. If you're searching frequently enough then we do indeed refresh every second but if there's a gap in searches of over 30s then we stop refreshing periodically and instead refresh just before the first search after the ≥30s gap, which can cause a surprising pause in how long that search takes.

File: docs/reference/search/near-real-time.asciidoc [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+[[near-real-time-configure-refresh]]
+[discrete]
+=== Configuring the refresh interval
```
Comment (DaveCTurner) [inline docs/reference/search/near-real-time.asciidoc:13]: I think we don't need this section, if we link to the `?refresh` docs above.

**Review by DaveCTurner — COMMENTED** [review]
I left some thoughts.

File: docs/reference/search/near-real-time.asciidoc (hunk 1 above)
Comment (lockewritesdocs) [reply docs/reference/search/near-real-time.asciidoc:2]: Agreed. I will migrate some content from the NRT page in the definitive guide to help bridge this gap.

File: docs/reference/search/near-real-time.asciidoc (hunk 3 above)
Comment (lockewritesdocs) [reply docs/reference/search/near-real-time.asciidoc:9]: Adding a list to explicitly mention those options, and removing "lightweight".

Comment (lockewritesdocs) [reply docs/reference/search/near-real-time.asciidoc:9]: Thanks for the clarification! I'm modifying this section to read:
>By default, Elasticsearch periodically refreshes indices every second, but only on indices that have received one search request or more in the last 30 seconds. This is why we say that Elasticsearch has _near_ real-time search: document changes are not visible to search immediately, but will become visible within this timeframe.

**Review by DaveCTurner — APPROVED** [review]
LGTM

**Pull request merged**